persistence/module.go

```diff
@@ -262,7 +263,7 @@ func (m *persistenceModule) shouldHydrateGenesisDb() (bool, error) {
 
 	blockHeight, err := readCtx.GetMaximumBlockHeight()
 	if err != nil {
-		return true, nil
+		return false, err
```

Comment:
I'm not super familiar with the logic here but from a cursory read of the context, my interpretation is that if there was an error loading the state from the DB then we assume that we "should hydrate" it, which is what (`persistendeModule#populateGenesisState()`)[https://github.com/pokt-network/pocket/blob/main/persistence/genesis.go#L15] seems to be responsible for. **The tests pass again locally when I revert this change.**

In a future refactor, I would expect that we may make some assertion about the error and have another branch which does return it if it is an unexpected error.